newIDE/app/public/external/piskel/piskel-main.js

```diff
@@ -281,87 +153,204 @@ function piskelCreateAnimation(pskl, piskelOptions) {
+
+  // Load a custom save file(s) header
+  const pathEditorHeaderDiv = document.getElementById('path-editor-header');
+  const headerStyle = {
+    saveFolderLabel: 'float: left;margin-left: 2px; font-size:15px;margin-top: 10px;color:aqua',
+    nameInput: 'font-family:"Courier New";height:27px;width:90px;float:left;margin-left: 2px;padding:4px;margin-top: 4px;font-size:15px;border: 2px solid #e5cd50;border-radius: 3px;background-color:black; color: #e5cd50;',
+    saveButton: 'float:right;margin-left:2px;margin-right:4px;border: 2px solid white;border-radius: 1px;margin-top: 5px;background-color:white;',
+    cancelButton: 'float:right;margin-left:2px;margin-right:2px;border: 2px solid white;border-radius: 1px;margin-top: 5px;background-color:white;',
+    setFolderButton: 'float:right;margin-left:2px;margin-right:2px;border: 2px solid white;border-radius: 1px;margin-top: 5px;background-color:white;',
+    fileExistsLabel: 'height:27px;color:aqua;float: left;margin-left: 2px;margin-top: 10px; font-size:15px;',
+  };
+  const savePathEditor = createPathEditorHeader({
+    parentElement: pathEditorHeaderDiv,
+    editorContentDocument: document,
+    onSaveToGd: saveToGD,
+    onCancelChanges: cancelChanges,
+    projectPath: receivedOptions.projectPath,
+    initialResourcePath: receivedOptions.resources[0] === undefined ?
+      '' : receivedOptions.resources[0].resourcePath,
+    headerStyle,
+    name: receivedOptions.name,
+    extension: piskelOptions.singleFrame ? '.png' : undefined,
+  });
+
+  if (!pskl) { return }
+
+  // Set piskel to tiled mode when editing a singleFrame object, set FPS from GD
+  pskl.UserSettings.set(pskl.UserSettings.SEAMLESS_MODE, piskelOptions.singleFrame);
+  const piskelController = pskl.app.piskelController;
+  piskelController.setFPS(piskelOptions.fps);
 
-  if (piskelOptions.resources.length === 0) {
+  // if no resources are being loaded, create a new animation
+  if (piskelOptions.resources.length === 0 && !piskelOptions.externalEditorData.pskl) {
     piskelCreateAnimation(pskl, piskelOptions);
     return;
   }
-  const imageData = [];
-  let maxWidth = -1;
-  let maxHeight = -1;
-  async.each(
-    piskelOptions.resources,
-    function(resource, callback) {
-      const image = new Image();
-      image.onload = function() {
-        imageData.push(image);
-        maxWidth = Math.max(image.width, maxWidth);
-        maxHeight = Math.max(image.height, maxHeight);
-        callback();
-      };
 
-      // onload will fire after `src` is set
-      try {
-        image.src = readBase64ImageFile(resource.resourcePath);
-      } catch (error) {
-        // Unable to load the image, ignore it.
-        console.error('Unable to load ', resource, error);
-        callback();
+  // If there is metadata, use it to load the frames with layers
+  // Note that metadata will be saved only if the user has more than one layers
+  if (piskelOptions.externalEditorData.pskl) {
```

Comment:
The comments that you put are useful, thanks for this!
The method is still quite long. Would it be possible to break it into smaller functions to enhance readibility?

Having a `if (piskelOptions. externalEditorData.pskl) {` followed by a lot of lines doing data transformation before the else is hard to read and hard to refactor for someone who is not familiar with the code. It's also harder to test if we want to add unit tests later.

Reply:
I will break it into functions, you're right it's getting long